sycl/source/detail/event_impl.cpp

```diff
@@ -116,8 +118,11 @@ static uint64_t inline getTimestamp() {
       .count();
 }
 
-const ur_event_handle_t &event_impl::getHandleRef() const { return MEvent; }
-ur_event_handle_t &event_impl::getHandleRef() { return MEvent; }
+ur_event_handle_t event_impl::getHandle() const { return MEvent.load(); }
+
+void event_impl::setHandle(const ur_event_handle_t UREvent) {
```

Comment:
Is there a reason not to do `void event_impl::setHandle(const ur_event_handle_t &UREvent)` ?

Reply:
Both ways are ok in this case because ur_event_handle_t is actually a pointer on its own: ur_event_handle_t_*.